### PR TITLE
feature(dropdown): add dropdown icon integration

### LINKED ISSUE #905: Dropdown Button with Icon
We decided to add icon attribute to the Dropdown Button component according to needs. 
Here you can see the [design documentation.](https://www.figma.com/design/RrcLH0mWpIUy4vwuTlDeKN/Baklava-Design-Guide?node-id=2158-8508&t=ADlTYVymgQ6F6mLV-1)

### PR DESCRIPTION
**Pull request by erbilnas**
Closes #905

### CODE REVIEW

File: src/components/select/bl-select.ts [hunk 1]
```diff
@@ -661,17 +661,26 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
...
+
+      if (option.textContent) {
+        try {
+          // Try locale-specific comparison first
+          optionText = option.textContent.toLocaleLowerCase(this.userLang);
+        } catch {
+          // Fallback to basic toLowerCase if locale is not supported
+          optionText = option.textContent.toLowerCase();
+        }
+      }
```
Comment (AykutSarac) [inline src/components/select/bl-select.ts:675]: why we're using try-catch here?

File: src/components/textarea/bl-textarea.test.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ describe("bl-textarea", () => {
           <div class="hint"></div>
         </div>
       `,
-      { ignoreAttributes: ["for", "id"] }
+      { ignoreAttributes: ["for", "id", "style"] }
```
Comment (AykutSarac) [inline src/components/textarea/bl-textarea.test.ts:33]: what is it for?

File: src/components/select/bl-select.ts (hunk 1 above)
Comment (erbilnas) [reply src/components/select/bl-select.ts:675]: It’s for code coverage. While I can't see the exact problem, if we don’t address it, the tests might behave unpredictably

File: src/components/textarea/bl-textarea.test.ts (hunk 2 above)
Comment (erbilnas) [reply src/components/textarea/bl-textarea.test.ts:33]: Different browsers interpret styles differently like margin:4px or margin: 4px. This inconsistency could cause error for the test. It came with the latest version of Playwright.

**Review by AykutSarac — APPROVED** [review]

**Review by fatihhayri — APPROVED** [review]

**Pull request merged**